chalice/app.py

```diff
@@ -146,6 +152,36 @@ class TooManyRequestsError(ChaliceViewError):
     TooManyRequestsError]
 
 
+class MultiDict(Mapping):
+    """A read only mapping of key to list of values.
+
+    Accessing it in the usual way will return the last value in the list.
+    Calling getlist will return a list of values with the same key.
+    """
+
+    def __init__(self, mapping):
+        self._dict = mapping or {}
+
+    def __getitem__(self, k):
+        lst = self._dict[k]
+        try:
+            return lst[-1]
+        except IndexError:
+            raise KeyError(k)
+
+    def getlist(self, k):
```

Comment:
I named it getlist because this is the name used by django and flask for the same method.
I can think of a better name, but it seems to me like it would be more intuitive to existing users like that.

Reply:
I see. If there is precedent then I'm fine with it.